py/server/deephaven/experimental/iceberg.py

```diff
@@ -980,15 +982,28 @@ def adapter(
     Args:
         name (Optional[str]): a descriptive name of the catalog; if omitted the catalog name is inferred from the
             catalog URI property.
-        properties (Optional[Dict[str, str]]): the properties of the catalog to load
-        hadoop_config (Optional[Dict[str, str]]): hadoop configuration properties for the catalog to load
+        properties (Optional[Dict[str, str]]): the properties of the catalog to load. By default, no properties are set.
+        hadoop_config (Optional[Dict[str, str]]): hadoop configuration properties for the catalog to load. By default,
+            no properties are set.
         s3_instructions (Optional[s3.S3Instructions]): the S3 instructions if applicable
+        enable_property_injection (bool): whether to enable Deephaven’s automatic injection of properties that work
+            around upstream issues and supply defaults needed for Deephaven’s Iceberg usage. Disable to manage all
+            properties yourself (not recommended). Defaults to `True`.
+
```

Comment:
What are the relationship between `properties` and `enable_property_injection`? How to resolve conflicts between user specified properties and auto-injected ones? Not sure if the question applies but some clarification could be helpful.

Also document the default value of `s3_instructions`?